### PR TITLE
Ensure that drawImage(detachedCanvas) throws

### PR DESCRIPTION
**Pull request by Kaiido**
See https://github.com/whatwg/html/issues/10026 and https://bugs.webkit.org/show_bug.cgi?id=266918

The specs are currently underspecified, though all implementations do agree that a detached `OffscreenCanvas` has a size of 0, which should throw when passed to `drawImage()`.  
Only Safari currently fails the test. A fix is on the way.

Note that I couldn't run `update-built` due to https://github.com/web-platform-tests/wpt/issues/23292 and hence I did create the generated html manually...

### CODE REVIEW

File: html/canvas/tools/yaml-new/drawing-images-to-the-canvas.yaml [hunk 1]
```diff
@@ -638,3 +638,11 @@
     @nonfinite ctx.drawImage(<bitmap>, <0 Infinity -Infinity NaN>, <0 Infinity -Infinity NaN>, <100 Infinity -Infinity NaN>, <50 Infinity -Infinity NaN>, <0 Infinity -Infinity NaN>, <0 Infinity -Infinity NaN>, <100 Infinity -Infinity NaN>, <50 Infinity -Infinity NaN>);
     @assert pixel 50,25 == 0,255,0,255;
   expected: green
+
+- name: 2d.drawImage.detachedcanvas
+  desc: drawImage with detached canvas as the source should throw exception
```
Comment (kenrussell) [inline html/canvas/tools/yaml-new/drawing-images-to-the-canvas.yaml:643]: Nit: this is really a detached OffscreenCanvas. Not sure whether more verbosity in the test name/description is warranted.

**Review by kenrussell — APPROVED** [review]
Looks fine to me. One small comment.

**Comment by Kaiido** [comment]
Small update/bump, the WebKit patch has landed a few weeks ago and now this test has been manually added there.
So now all major vendors pass it. (And I don't have write access so I need help for merging.)

**Comment by annevk** [comment]
Thanks @Kaiido for your continued efforts on improving canvas!

**Pull request merged**